### PR TITLE
Fix USB Bluetooth (HCI SPI)

### PR DESCRIPTION
**Pull request by loicpoulain**
This PR:
- adds TX thread to bluetooth class in order call bt_send in thread context.
- Fixes assert/mem corrupt issue due to unreserved headroom (for BT packet H4 type)
- Introduces transfer API for ACL ep out

### CODE REVIEW

**Review by finikorg — COMMENTED** [review]
Looks OK

**Comment by loicpoulain** [comment]
@jfischer-phytec-iot, could you please review.

File: subsys/usb/class/bluetooth.c [hunk 1]
```diff
@@ -120,34 +88,31 @@ static void hci_tx_thread(void)
...
 
-	if (!len || len > BT_BUF_ACL_SIZE) {
-		SYS_LOG_ERR("Incorrect length: %u\n", len);
-		return;
+	if (buf) {
+		net_buf_unref(buf);
 	}
 
 	buf = net_buf_alloc(&acl_tx_pool, K_NO_WAIT);
 	if (!buf) {
```
Comment (jfischer-phytec-iot) [inline subsys/usb/class/bluetooth.c]: Is not it issued each time it is called from USB_DC_CONFIGURED?  `if (!buf && size)` ?

**Review by jfischer-no — COMMENTED** [review]
LGTM except a71fd69199b75f6c615e51a0a7d4d83464574654, I would like to take a closer look

**Review by jfischer-no — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Pull request merged**